### DISCUSSION TITLE
Automatically checking for dead links and removing or "silencing" them?

### DISCUSSION THREAD

**Discussion by sxflynn** [iptv-org/community#496] [Ideas]
There are thousands of m3u links in the playlist, and many of them are dead links.

Would it be possible to setup a script that checks for dead URLs and automatically removes those from the playlist? Or if you are concerned with deleting something, maybe sets a new JSON flag to "dead" = true?

That way we have the ability to download an m3u playlist that is filled with links that have been recently checked by an automated script as being good links that work. It's a huge bummer to channel surf and so many of them are dead.

**Comment by Alstruit** [comment]
We already tried that with Actions and the repo was suspended due to excessive resources. So that is out of the question. You may try to check the links on locally.

**Reply by sxflynn** [reply]
I have been making a python script that does this. Perhaps a best-quality script should be added to the repo to simplify this process?

**Comment by PopeyeTheSai10r** [comment]
[playlist:validate](https://github.com/iptv-org/iptv/blob/master/CONTRIBUTING.md#:~:text=playlist%3Avalidate%3A%20%D1%81hecks%20ids%20and%20links%20in%20internal%20playlists%20for%20errors) may help with this. Give it a try and let us know how it goes.
[iptv-checkerr](https://github.com/freearhey/iptv-checker) can be another option.

**Comment by freearhey** [comment]
https://github.com/iptv-org/iptv/blob/master/CONTRIBUTING.md#how-to-find-a-broken-stream
